tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs

```diff
@@ -52,9 +52,16 @@ public static void Generate(string sourcePath, string outputPath, ConsoleLogger
                                 logger.LogMessage($"Found extension: {foundRef.TypeName}");
                             }
                         }
-                        catch (Exception exc)
+                        catch (FileNotFoundException ex)
                         {
-                            logger.LogError($"Could not evaluate '{Path.GetFileName(path)}' for extension metadata. If this assembly contains a Functions extension, ensure that all dependent assemblies exist in '{sourcePath}'. If this assembly does not contain any Functions extensions, this message can be ignored. Exception message: {exc.Message}");
+                            // Don't log this as an error. This will almost always happen due to some publishing artifacts (i.e. Razor) existing
+                            // in the functions bin folder without all of their dependencies present. These will almost never have Functions extensions,
+                            // so we don't want to write out errors every time there is a build. This message can be seen with detailed logging enabled.
+                            logger.LogMessage($"Could not evaluate '{Path.GetFileName(path)}' for extension metadata. If this assembly contains a Functions extension, ensure that all dependent assemblies exist in '{sourcePath}'. If this assembly does not contain any Functions extensions, this message can be ignored. Exception message: {ex.Message}");
```

Comment:
I decided to make this a message b/c it's almost never going to be an issue so it felt wrong to constantly log warnings to people that had builds set up this way. This allows it to be discoverable (by turning on detailed msbuild logging) yet not constantly warning.